src/Features/VisualBasic/Portable/Structure/Providers/SelectBlockStructureProvider.vb

```diff
@@ -19,4 +19,46 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Structure
                                type:=BlockTypes.Conditional, isCollapsible:=True))
         End Sub
     End Class
+
+    Friend Class CaseStatementSyntaxStructureProvider
+        Inherits AbstractSyntaxNodeStructureProvider(Of CaseStatementSyntax)
+
+        Protected Overrides Sub CollectBlockSpans(node As CaseStatementSyntax,
+                                                  spans As ArrayBuilder(Of BlockSpan),
+                                                  options As OptionSet,
+                                                  cancellationToken As CancellationToken)
+            Dim _case_ = node.Cases.FirstOrDefault
+            If _case_ IsNot Nothing Then
+
+                Dim banner As String = String.Empty
+
+                If TypeOf _case_ Is SimpleCaseClauseSyntax Then
+                    Dim simple = DirectCast(_case_, SimpleCaseClauseSyntax)
+                    banner = $"Case {simple.Value}"
+                ElseIf TypeOf _case_ Is RangeCaseClauseSyntax Then
+                    Dim range = DirectCast(_case_, RangeCaseClauseSyntax)
+                    banner = $"Case {range.LowerBound} To {range.UpperBound}"
+                ElseIf TypeOf _case_ Is RelationalCaseClauseSyntax Then
+                    Dim relational = DirectCast(_case_, RelationalCaseClauseSyntax)
+                    banner = $"Case {relational.Value}"
+                ElseIf TypeOf _case_ Is ElseCaseClauseSyntax Then
+                    Dim relational = DirectCast(_case_, ElseCaseClauseSyntax)
+                    banner = $"Case {relational.ElseKeyword.Text}"
```

Comment:
do we need all of this?  I don't believe we generally synthesize the banner text.  It's created directly from the node in question.

Reply:
This comment still applies.

Reply:
VB allows the multiple case clauses to be appear over multiple lines.
![additioncollapsibleblocks5](https://user-images.githubusercontent.com/900107/41320515-0295ef82-6e98-11e8-94e6-d2baf180a658.png)
**When Collapsed**
![additioncollapsibleblocks6](https://user-images.githubusercontent.com/900107/41320516-03dee268-6e98-11e8-9b2a-c56356ae50ec.png)
So the just using the first clause and indicating it also has additional clauses with `, ...` solves this.

Reply:
> VB allows the multiple case clauses to be appear over multiple lines.

Sure.  But that's true of most of the nodes we already allow you to collapse.  For example:

![image](https://user-images.githubusercontent.com/4564579/41321019-e92c8268-6e56-11e8-826e-71380b9e06c1.png)

This collapses to:

![image](https://user-images.githubusercontent.com/4564579/41321030-f37f628a-6e56-11e8-96af-37d27d069827.png)

Our code already properly handles this, and there's no need to manually construct (or pick apart) things as you are doing here...